tests/Aspire.Hosting.Azure.Tests/ProvisioningTestHelpers.cs

```diff
@@ -242,4 +241,182 @@ public Task<UserPrincipal> GetUserPrincipalAsync(CancellationToken cancellationT
 internal sealed class TestTokenCredentialProvider : ITokenCredentialProvider
 {
     public TokenCredential TokenCredential => new TestTokenCredential();
+}
+
+/// <summary>
+/// Test implementation of SubscriptionResource using Azure SDK testing patterns.
+/// </summary>
+internal sealed class TestSubscriptionResource : SubscriptionResource
+{
+    // Use public constructor for testing
+    public TestSubscriptionResource()
+    {
+    }
+
+    // Override Data property to return test data
+    public override SubscriptionData Data => TestSubscriptionData.Instance;
+}
+
+/// <summary>
+/// Test implementation of ResourceGroupResource using Azure SDK testing patterns.
+/// </summary>
+internal sealed class TestResourceGroupResource : ResourceGroupResource
+{
+    // Use public constructor for testing
+    public TestResourceGroupResource()
+    {
+    }
+
+    // Override Data property to return test data
+    public override ResourceGroupData Data => TestResourceGroupData.Instance;
+}
+
+/// <summary>
+/// Test implementation of TenantResource using Azure SDK testing patterns.
+/// </summary>
+internal sealed class TestTenantResource : TenantResource
+{
+    // Use public constructor for testing
+    public TestTenantResource()
+    {
+    }
+
+    // Override Data property to return test data
+    public override TenantData Data => TestTenantData.Instance;
+}
+
+/// <summary>
+/// Test data for SubscriptionResource using reflection to create instances.
+/// </summary>
+internal static class TestSubscriptionData
```

Comment:
See https://github.com/Azure/azure-sdk-for-net/blob/9ac52a165101b727ff606dd50dd82f771ec7c710/sdk/core/Azure.Core/README.md#mocking

Also https://learn.microsoft.com/en-us/dotnet/api/azure.resourcemanager.models.resourcemanagermodelfactory.subscriptiondata?view=azure-dotnet